src/field/pokemon.ts

```diff
@@ -3163,13 +3161,21 @@ export default abstract class Pokemon extends Phaser.GameObjects.Container {
    * @param ignoreFaintPhase boolean to ignore adding a FaintPhase, passsed to damage()
    * @returns integer of damage done
    */
-  damageAndUpdate(damage: number, result?: DamageResult, critical: boolean = false, ignoreSegments: boolean = false, preventEndure: boolean = false, ignoreFaintPhase: boolean = false, source?: Pokemon): number {
+  damageAndUpdate(damage: number, result?: DamageResult, critical: boolean = false, ignoreSegments: boolean = false, ignoreFaintPhase: boolean = false, source?: Pokemon): number {
```

Comment:
It's probably a good idea to change the optional parameters to be an object, ie:
```suggestion
  damageAndUpdate(damage: number,
    {
      result = HitResult.EFFECTIVE,
      critical = false,
      ignoreSegments = false,
      ignoreFaintPhase = false,
    }:
    {
      result?: DamageResult,
      critical?: boolean,
      ignoreSegments?: boolean,
      ignoreFaintPhase?: boolean,
      source?: Pokemon,
     }
  ): number {
```

Reply:
Refactored in [855d795](https://github.com/pagefaultgames/pokerogue/pull/5182/commits/855d795f2d4774423ff6e19808c7d4ead0e11a97)